admin-tools/scripts/src/main/scala/com/gu/mediaservice/ImagesGroupByProgressState.scala

```diff
@@ -0,0 +1,58 @@
+package com.gu.mediaservice
+
+import com.gu.mediaservice.indexing.{IndexInputCreation, ProduceProgress}
+import com.typesafe.scalalogging.LazyLogging
+
+object ImagesGroupByProgressState extends App with LazyLogging {
+
+  if (args.isEmpty) throw new IllegalArgumentException("please provide dynamo table name")
+
+  import InputIdsStore._
+
+  private val dynamoTable = args(0)
+  private val ddbClient = BatchIndexHandlerAwsFunctions.buildDynamoTableClient(dynamoTable)
+  private val stateIndex = ddbClient.getIndex(StateField)
+
+  def execute() = {
+
+    def stateNameToCount(progressType: ProduceProgress) = {
+      logger.info(s"calculating stateNameToCount for $progressType")
+      val result = stateIndex.query(getAllMediaIdsWithinStateQuery(progressType.stateId))
+      val iterator = result.iterator()
+      var count: Long = 0
+      while (iterator.hasNext) {
+        count += 1
+        iterator.next()
+      }
+      val ans = progressType.name -> count
```

Comment:
This is quite Java-esque

```suggestion
      val ans = progressType.name -> result.iterator.asScala.length
```
seems better? (You'll also need to `import scala.collection.JavaConverters._`)

Reply:
god point, looks much better now